File: nysol_python/mcmd/methods/convTex/db/smp/func_age.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='age'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,dob
1,19641010
2,20000101
3,
4,19770812
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
2013年9月1日における年令を求める。
'''
script['sh_code']='''
mcal c='age($d{dob},0d20130901)' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='age($d{dob},0d20130901)', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_age.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,dob
1,19641010
2,20000101
3,
4,19770812
EOF
)}


############## Example 1
title="Basic Example"
comment=<<'EOF'
Compute the age from date of birth to a fixed point on calendar on September 1, 2013.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='age($d{dob},0d20130901)' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

